From 6eb5bc9f70957ff1f3edd87dbc81e6909468cc09 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sun, 21 Jul 2024 20:13:01 +0200
Subject: [PATCH] arm64: dts: imx8mn-solidsense-n8: set bluetooth max baudrate
 to 3Mbps

Murata LBEE5HY1MW Bluetooth component theoretically supports up to 4Mbps
baudrate. With current drivers only 3Mbps is usable.

Set maximum speed to 3M.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../boot/dts/freescale/imx8mn-solidsense-n8-compact.dts      | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mn-solidsense-n8-compact.dts b/arch/arm64/boot/dts/freescale/imx8mn-solidsense-n8-compact.dts
index 3658317d9..d38e0334b 100644
--- a/arch/arm64/boot/dts/freescale/imx8mn-solidsense-n8-compact.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mn-solidsense-n8-compact.dts
@@ -736,6 +736,9 @@ &uart1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart1_pins>;
 	uart-has-rtscts;
+	/* select 80MHz parent clock to support maximum baudrate 4Mbps */
+	assigned-clocks = <&clk IMX8MN_CLK_UART1>;
+	assigned-clock-parents = <&clk IMX8MN_SYS_PLL1_80M>;
 	status = "okay";
 
 	bluetooth {
@@ -743,7 +746,7 @@ bluetooth {
 		device-wakeup-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
 		host-wakeup-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
 		shutdown-gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
-		max-speed = <1500000>;
+		max-speed = <3000000>;
 	};
 };
 
-- 
2.43.0

